packages/xstate-react/src/useInterpret.ts

```diff
@@ -82,6 +82,12 @@ export function useIdleInterpreter(
     Object.assign(service.machine.options.delays!, delays);
   }, [actions, guards, activities, services, delays]);
 
+  useIsomorphicLayoutEffect(() => {
+    if (options.subscribe) {
+      return service.subscribe(options.subscribe).unsubscribe;
+    }
+  }, []);
```

Comment:
this suffers from the stale closure problem, it should be:
```suggestion
  }, [options.subscribe]);
```